pkg/kubelet/status/manager.go

```diff
@@ -382,6 +382,11 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 				glog.V(3).Infof("Pod %q is terminated, but some containers are still running", format.Pod(pod))
 				return
 			}
+			if pod.DeletionGracePeriodSeconds != nil && *pod.DeletionGracePeriodSeconds == 0 {
```

Comment:
How are we guaranteeing the apiserver deletes it?  The APIserver can crash between when it registers the deletion 0 and when it would delete, and it has no logic to clean up on recovery.  The kubelet ultimately owns the responsibility for deleting the container.  At a minimum, the Kubelet has to continue to own the responsibility to ensure it's fully deleted.